pkg/asset/agent/image/ignition.go

```diff
@@ -94,30 +104,17 @@ func (a *Ignition) Generate(dependencies asset.Parents) error {
 	agentManifests := &manifests.AgentManifests{}
 	agentConfigAsset := &agentconfig.AgentConfig{}
 	extraManifests := &manifests.ExtraManifests{}
-	dependencies.Get(agentManifests, agentConfigAsset, extraManifests)
+	ignitionBaseAsset := &IgnitionBase{}
+	dependencies.Get(agentManifests, agentConfigAsset, extraManifests, ignitionBaseAsset)
 
 	pwd := &password.KubeadminPassword{}
 	dependencies.Get(pwd)
 	pwdHash := string(pwd.PasswordHash)
 
 	infraEnv := agentManifests.InfraEnv
 
-	config := igntypes.Config{
-		Ignition: igntypes.Ignition{
-			Version: igntypes.MaxVersion.String(),
-		},
-		Passwd: igntypes.Passwd{
-			Users: []igntypes.PasswdUser{
-				{
-					Name: "core",
-					SSHAuthorizedKeys: []igntypes.SSHAuthorizedKey{
-						igntypes.SSHAuthorizedKey(infraEnv.Spec.SSHAuthorizedKey),
-					},
-					PasswordHash: &pwdHash,
-				},
-			},
-		},
-	}
+	config := ignitionBaseAsset.Config
```

Comment:
This doesn't do a deep copy, so I think it will end up modifying the content of the ignitionBase. I'm not sure how the asset store will respond to that, but I think things may get weird (so if you ran e.g. `agent create image` followed by `agent create unconfigured-ignition`, the resulting ignition file may in fact be a partly-configured one).

In the absence of a DeepCopy() method, you could dump to JSON and then reload.

An alternative you may already have considered is to use golang struct composition instead of the asset graph to incorporate the base implementation. (That would require adding a separate asset for the unconfigured ignition, since that one is a WritableAsset but we don't want this one to be.)

Reply:
Ah I see where this may be confusing. Ignition.Config is a pointer while IgnitionBase.Config is not. So this line should create a copy instead of modifying the content of the IgnitionBase. I'll add a unit test to verify that the Ignition.Generate doesn't change the IgnitionBase.

Reply:
That is confusing, but the real problem is that it's only a shallow copy. So while any pointers, maps, or slices will get copied (along with other fields like strings/ints and structs), the stuff they are pointing to will not.
so e.g. on the line below this, `config.Passwd.Users` is a new copy of the slice, but setting `config.Passwd.Users[0].PasswordHash` modifies the original object in the underlying array.

Reply:
That is tricky. Ok I will copy function to dump and reload from yaml.